### PR TITLE
Certificate offboarding

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

File: api/src/Security/CertificateAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            ->where($qb->expr()->like('a.certificates', $qb->expr()->literal("%$certificate%")));
+
+        $application = $qb->getQuery()->disableResultCache()->getOneOrNullResult();
+
+        if ($application instanceof Application === true) {
+            return $application;
+        }
+
+        throw new AuthenticationException('No application found for certificate');
+    }
```
Comment (rubenvdlinde) [inline api/src/Security/CertificateAuthenticator.php:68]: //end class comment missing

File: api/src/Security/CertificateAuthenticator.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            && $request->server->get('SSL_CLIENT_S_DN') !== null)
+            || ($request->headers->has('SSL_CLIENT_VERIFY') !== false
+            && $request->headers->has('SSL_CLIENT_CERT') !== false
+            && $request->headers->has('SSL_CLIENT_SUBJECT_DN') !== false
+            && $request->headers->get('SSL_CLIENT_VERIFY') !== ''
+            && $request->headers->get('SSL_CLIENT_CERT') !== ''
+            && $request->headers->get('SSL_CLIENT_SUBJECT_DN') !== '');
+    }
+
+    private function findApplicationByCertificate(string $certificate): ?Application
```
Comment (rubenvdlinde) [inline api/src/Security/CertificateAuthenticator.php:52]: docblock mising

File: api/migrations/Version20230221130043.php [hunk 3]
```diff
@@ -0,0 +1,373 @@
+<?php
```
Comment (rubenvdlinde) [inline api/migrations/Version20230221130043.php:1]: Did you mean to make a migration?

**Review by rubenvdlinde — CHANGES_REQUESTED** [review]
See commetns, please make sure you need everyting that you include in your migration (besides certifactes on applications)

And have a look at the doc blocks and comments in thet authenticator

File: api/src/Security/CertificateAuthenticator.php (hunk 2 above)
Comment (rjzondervan) [reply api/src/Security/CertificateAuthenticator.php:52]: Will fix

File: api/src/Security/CertificateAuthenticator.php (hunk 1 above)
Comment (rjzondervan) [reply api/src/Security/CertificateAuthenticator.php:68]: Will fix

File: api/migrations/Version20230221130043.php (hunk 3 above)
Comment (rjzondervan) [reply api/migrations/Version20230221130043.php:1]: Yes, the application is changed to contain certificates, which needs a database update

**Review by rubenvdlinde — APPROVED** [review]

**Pull request merged**